regolith/mongoclient.py

```diff
@@ -389,13 +389,9 @@ def all_documents(self, collname, copy=True):
     def insert_one(self, dbname, collname, doc):
         """Inserts one document to a database/collection."""
         doc = doc_cleanup(doc)
-        try:
-            valid = validate_doc(collname, doc, self.rc)
-            if not valid:
-                sys.exit("Validation failed. Upload cancelled... exiting program")
-        except Exception as e:
-            print(e)
-            sys.exit("Validation failed unexpectedly. Upload cancelled... exiting program")
+        valid = validate_doc(collname, doc, self.rc)
+        if not valid:
+            raise ValueError("Validation failed. Upload cancelled.")
```

Comment:
did you not want to return the collname and doc that was invalid for a more useful error message?

Reply:
My most recent commit addresses this and adds news. It doesn't seem to be auto-testing yet though.